tools/check_release_notes.sh

```diff
@@ -0,0 +1,178 @@
+            token_path="$2"
+            token="$(cat "$token_path")"
+            shift 2
+            ;;
+        --token)
+            token="$2"
+            tmp_token="$(mktemp -t token-XXXXXXXXXX)"
+            echo "$token" >"$tmp_token"
+            token_path="$tmp_token"
+            shift 2
+            ;;
+        --org)
+            REPO_OWNER="$2"
+            shift 2
+            ;;
+        --repo)
+            REPO_NAME="$2"
+            shift 2
+            ;;
+        --pr)
+            PULL_NUMBER="$2"
+            shift 2
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            print_error_and_exit "unknown option: $1"
+            ;;
+        esac
+    done
+}
+
+#This script relies on the REPO_OWNER, REPO_NAME, and PULL_NUMBER environment
+#variables as defined in https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables.
+validate_opts() {
+    if [ -z "${REPO_OWNER:-}" ]; then
+        print_error_and_exit "REPO_OWNER is a required option. It must match the GitHub org."
+        exit 1
+    fi
+
+    if [ -z "${REPO_NAME:-}" ]; then
+        print_error_and_exit "REPO_NAME is a required option. It must match the GitHub repo."
+        exit 1
+    fi
+
+    if [ -z "${PULL_NUMBER:-}" ]; then
+        print_error_and_exit "PULL_NUMBER is a required option. It must match the GitHub pull request number."
+        exit 1
+    fi
+}
+
+# Curl the GitHub API to get a list of files for the specified PR. If files are
+# found, exit. We might eventually want to validate the data here.
+checkForFiles() {
+    echo "Requesting files from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER}"
+    if [ -z "${token}" ]; then
+        ghFiles=$(curl -L -s --show-error --fail \
+            --header "Accept: application/vnd.github.v3+json" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
+    else
+        ghFiles=$(curl -L -s --show-error --fail \
+            --header "Accept: application/vnd.github.v3+json" \
+            --header "Authorization: Bearer ${token}" \
+            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
+    fi
+
+    #if return code isn't 0, print the error and exit.
+    #shellcheck disable=SC2181
+    if [ $? != 0 ]; then
+        exit 1
+    fi
+
+    releaseNotesFiles=$(echo "${ghFiles}" | jq 'map(.filename)' |
+            grep 'releasenotes\/notes\/.*\.yaml' )
+
+    if [ -z "${releaseNotesFiles}" ]; then
+        echo "No release notes files found."
+    else
+        echo "Found release notes entries"
+        exit 0
+    fi
+}
+
+checkForLabel() {
+    # If no release notes files were found in the PR, go back and request the actual
+    # PR info (instead of the files) and check the labels.
+    echo "Requesting labels from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER}"
+
+    if [ -z "${token}" ]; then
```

Comment:
I added the token because of the rate limiter: 
```
For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.
```

Which would mean roughly 30 pull requests checks per IP address per hour. Not sure how fast we'd hit this but I'd assume pretty quickly.

Reply:
https://developer.github.com/v3/